src/test/java/fi/helsinki/cs/tmc/core/commands/SubmitAdaptiveExerciseToSkillifierTest.java

```diff
@@ -0,0 +1,125 @@
+
+import fi.helsinki.cs.tmc.core.communication.TmcServerCommunicationTaskFactory;
+import fi.helsinki.cs.tmc.core.configuration.TmcSettings;
+import fi.helsinki.cs.tmc.core.domain.Course;
+import fi.helsinki.cs.tmc.core.domain.Exercise;
+import fi.helsinki.cs.tmc.core.domain.ProgressObserver;
+import fi.helsinki.cs.tmc.core.domain.submission.SubmissionResult;
+import fi.helsinki.cs.tmc.core.holders.TmcLangsHolder;
+import fi.helsinki.cs.tmc.core.holders.TmcSettingsHolder;
+import fi.helsinki.cs.tmc.core.utils.MockSettings;
+import fi.helsinki.cs.tmc.core.utils.TestUtils;
+import fi.helsinki.cs.tmc.langs.util.TaskExecutor;
+import fi.helsinki.cs.tmc.langs.util.TaskExecutorImpl;
+
+import com.google.common.base.Optional;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.concurrent.Callable;
+
+
+public class SubmitAdaptiveExerciseToSkillifierTest {
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+    @Mock
+    ProgressObserver mockObserver;
+    @Spy
+    TmcSettings settings = new MockSettings();
+    @Mock
+    TmcServerCommunicationTaskFactory factory;
+    @Mock
+    Course mockCourse;
+    @Mock
+    Exercise mockExercise;
+
+    private static final URI PASTE_URI = URI.create("http://example.com/paste");
+    private static final URI SUBMISSION_URI = URI.create("http://example.com/submission");
+    private static final TmcServerCommunicationTaskFactory.SubmissionResponse STUB_RESPONSE =
+            new TmcServerCommunicationTaskFactory.SubmissionResponse(SUBMISSION_URI, PASTE_URI);
+
+    private static final String STUB_PROSESSING_ERRORED_RESPONSE = "{error: \"failed to submit the exercise\"}";
+    private static final String STUB_PROSESSING_DONE_RESPONSE = "{status: \"OK\"}";
+
+    private Command<SubmissionResult> command;
+    private Path arithFuncsTempDir;
+    private TaskExecutor langs;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        TmcSettingsHolder.set(settings);
+
+        langs = spy(new TaskExecutorImpl());
+        TmcLangsHolder.set(langs);
+        TmcSettingsHolder.set(settings);
+        Exercise ex = new Exercise("Osa02_01.WilliamLovelace", "Example");
+        command = new SubmitAdaptiveExerciseToSkillifier(mockObserver, ex, factory);
+
+        arithFuncsTempDir = TestUtils.getProject(this.getClass(), "arith_funcs");
+        when(mockExercise.getExerciseDirectory(any(Path.class))).thenReturn(arithFuncsTempDir);
+        when(settings.getLocale()).thenReturn(new Locale("FI"));
+        Optional<String> mockToken = Optional.of("testToken");
+        settings.setToken(mockToken);
+
+        when(factory.getSkillifierUrl(anyString())).thenReturn(URI.create("www.someurl.com"));
```

Comment:
Prefer "www.example.com" as it's reserved by RFC2606. Using anything under the ".test" TLD is also fine as it's similarly reserved.